gateway/reverse_proxy.go

```diff
@@ -538,21 +538,39 @@ func (p *ReverseProxy) ServeHTTPForCache(rw http.ResponseWriter, req *http.Reque
 	return resp
 }
 
+// This returns the expected 30 second default timeout in case
+// the ProxyDefaultTimeout value is undefined.
```

Comment:
```suggestion
// proxyDefaultTimeout returns the default timeout of 30 seconds
// if the global ProxyDefaultTimeout value is undefined.
```

Reply:
added suggestion for GoDoc rule starting with func name and also some improvements

Reply:
godoc applies to public functions